### PR TITLE
[SPARK-15594][SQL] ALTER TABLE SERDEPROPERTIES does not respect partition spec

### PR DESCRIPTION
**Pull request by andrewor14**
## What changes were proposed in this pull request?

These commands ignore the partition spec and change the storage properties of the table itself:

```
ALTER TABLE table_name PARTITION (a=1, b=2) SET SERDE 'my_serde'
ALTER TABLE table_name PARTITION (a=1, b=2) SET SERDEPROPERTIES ('key1'='val1')
```

Now they change the storage properties of the specified partition.
## How was this patch tested?

DDLSuite

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala [hunk 1]
```diff
@@ -871,6 +879,58 @@ class DDLSuite extends QueryTest with SharedSQLContext with BeforeAndAfterEach {
...
+      assert(catalog.getPartition(tableIdent, spec).storage.serdeProperties.isEmpty)
+      sql("ALTER TABLE dbx.tab1 PARTITION (a=1, b=2) SET SERDE 'org.apache.madoop' " +
+        "WITH SERDEPROPERTIES ('k' = 'v', 'kay' = 'vee')")
+      assert(catalog.getPartition(tableIdent, spec).storage.serde == Some("org.apache.madoop"))
+      assert(catalog.getPartition(tableIdent, spec).storage.serdeProperties ==
+        Map("k" -> "v", "kay" -> "vee"))
+    }
+    // set serde properties only
+    sql("ALTER TABLE dbx.tab1 PARTITION (a=1, b=2) " +
+      "SET SERDEPROPERTIES ('k' = 'vvv', 'kay' = 'vee')")
```
Comment (yhuai) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala]: Do data source tables allow this?

Comment (andrewor14) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala]: right now they do. Should they not?

Comment (yhuai) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala]: For data source tables, seems we do not store any partition specs. I am wondering why metastore does not complain?

Comment (andrewor14) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala]: ah I see. We can add 1 more check

**Comment by yhuai** [comment]
lgtm. Merging to master and branch 2.0.

**Pull request closed without merging**